DeepSpeech.ipynb

```diff
@@ -1021,80 +1177,158 @@
-    "                \n",
-    "        # Prepare tensor board logging\n",
-    "        merged = tf.merge_all_summaries()\n",
-    "        writer = tf.train.SummaryWriter('%s/%s' % (\"logs\", time.strftime(\"%Y%m%d-%H%M%S\")), session.graph)\n",
+    "        # Loop over the batches\n",
+    "        for batch in range(total_batch/len(available_devices)):\n",
+    "            # Compute the average loss for the last batch\n",
+    "            _, batch_avg_loss = session.run([apply_gradient_op, tower_loss])\n",
+    "\n",
+    "            # Add batch to total_accuracy\n",
+    "            total_accuracy += session.run(accuracy)\n",
+    "\n",
+    "            # Log all variable states in current step\n",
+    "            step = epoch * total_batch + batch * len(available_devices)\n",
+    "            summary_str = session.run(merged)\n",
+    "            writer.add_summary(summary_str, step)\n",
+    "            writer.flush()\n",
     "        \n",
-    "        # Loop over the data set for training_epochs epochs\n",
-    "        for epoch in range(training_iters):\n",
-    "            # Define total accuracy for the epoch\n",
-    "            total_accuracy = 0\n",
-    "            \n",
-    "            # Loop over the batches\n",
-    "            for batch in range(total_batch/len(available_devices)):\n",
-    "                # Compute the average loss for the last batch\n",
-    "                _, batch_avg_loss = session.run([apply_gradient_op, tower_loss])\n",
-    "                \n",
-    "                # Add batch to total_accuracy\n",
-    "                total_accuracy += session.run(accuracy)\n",
-    "                \n",
-    "                # Log all variable states in current step\n",
-    "                step = epoch * total_batch + batch * len(available_devices)\n",
-    "                summary_str = session.run(merged)\n",
-    "                writer.add_summary(summary_str, step)\n",
-    "                writer.flush()\n",
-    "\n",
-    "            # Print progress message\n",
-    "            if epoch % display_step == 0:\n",
-    "                print \"Epoch:\", '%04d' % (epoch+1), \"avg_cer=\", \"{:.9f}\".format((total_accuracy / total_batch))\n",
-    "            \n",
-    "            # Checkpoint the model\n",
-    "            if (epoch % checkpoint_step == 0) or (epoch == training_iters - 1):\n",
-    "                checkpoint_path = os.path.join(checkpoint_dir, 'model.ckpt')\n",
-    "                print \"Checkpointing in directory\", \"%s\" % checkpoint_dir\n",
-    "                saver.save(session, checkpoint_path, global_step=epoch)\n",
+    "        # Print progress message\n",
+    "        if epoch % display_step == 0:\n",
+    "            print \"Epoch:\", '%04d' % (epoch+1), \"avg_cer=\", \"{:.9f}\".format((total_accuracy / total_batch))\n",
+    "            _, last_train_wer = print_wer_report(session, \"Training\", tower_decodings, tower_labels)\n",
```

Comment:
According to the example of [[1](https://www.tensorflow.org/versions/r0.9/tutorials/mnist/pros/index.html#train-and-evaluate-the-model)], this will produce incorrect results as it is printing results from the training data where `dropout_rate=0.05` and not `dropout_rate=0.0`.

This is due to the fact that the line
```
tower_decodings, tower_labels, tower_gradients, tower_loss, accuracy =
    get_tower_results(n_steps, data_sets.train, optimizer)
```
used to create `tower_decodings`, `tower_labels`, `tower_gradients`, `tower_loss`, and `accuracy` passes an `optimizer` in which causes `dropout_rate=0.05` to be used. 

It seems like it might make more sense to introduce something like..
```
train_tower_decodings, train_tower_labels = decode_batch(data_sets.train)
```
then print the results from this like you do for the validation data.